modules/manifold/src/utils/data-processor.js

```diff
@@ -252,22 +260,23 @@ export function columnsAndFieldsFromYPred(yPred, classLabels) {
  * @param {Array<Number|String>} yTrue ground truth
  * @return {Object} {columns - array of array of columns; fields: array of field metadata} in the same order
  */
-export function columnsAndFieldsFromYTrue(yTrue) {
+export function columnsAndFieldsFromYTrue(yTrue, nClasses) {
   return {
     columns: [yTrue],
     fields: [
       {
         name: GROUND_TRUTH_NAME,
-        type: FIELD_TYPE.GROUND_TRUTH,
-        dataType: typeof yTrue[0] === 'string' ? 'string' : 'float',
+        type: nClasses < 2 ? FEATURE_TYPE.NUMERICAL : FEATURE_TYPE.CATEGORICAL,
+        role: FIELD_ROLE.GROUND_TRUTH,
+        dataType: computeDataType(yTrue),
       },
     ],
   };
 }
 
 /**
  * compute model score columns given yPred, yTrue columns
- * @param {Array<Array<Number|String>} yPred prediction columns, from all models and all classes
+ * @param {Array<Array<Number>} yPred prediction columns, from all models and all classes
```

Comment:
could you explain this change?

Reply:
It was a mistake before, our documentation https://github.com/uber/manifold#upload-csv-to-demo-app maintains that values in `yPred` should always be numbers